icu4c/source/common/unicode/unistr.h

```diff
@@ -1924,30 +1924,39 @@ class U_COMMON_API UnicodeString : public Replaceable
   inline UBool isBogus() const;
 
 #ifndef U_HIDE_DRAFT_API
+private:
+  // These type aliases are private; there is no guarantee that they will remain
+  // aliases to the same types in subsequent versions of ICU.
+  // Note that whether `std::u16string_view::const_iterator` is a pointer or a
+  // class that models contiguous_iterator is platform-dependent.
+  using iterator = std::u16string_view::const_iterator;
```

Comment:
@markusicu This type alias is very much not part of the API, but let me know if I should give it a scarier name (internal_iterator? unspecified_iterator? unstable_iterator? you_should_feel_bad_if_you_depend_on_the_type_of_this_iterator?)

Reply:
unspecified_iterator sgtm